syntax = "proto3";

package banditpylib;

// Next tag: 2
message Arm {
  // The arm id is an integer
  int32 id = 1;
}

// Next tag: 3
message ArmPullsPair {
  // Arm to be pulled
  Arm arm = 1;
  // Number of times to pull
  int32 pulls = 2;
}

// Next tag: 2
message Actions {
  repeated ArmPullsPair arm_pulls_pairs = 1;
}

// Next tag: 3
message ArmRewardsPair {
  // Arm to be pulled
  Arm arm = 1;
  // The corresponding rewards
  repeated float rewards = 2;
}

// Next tag: 2
message Feedback {
  repeated ArmRewardsPair arm_rewards_pairs = 1;
}

// Next tag: 6
message DataItem {
  // Bandit name
  string bandit = 1;
  // Learner name
  string learner = 2;
  // Total communication rounds which is
  int32 rounds = 3;
  // Total actions performed by the learner
  int32 total_actions = 4;
  // Learner's regret i.e., the difference between the optimal policy and the
  // one generated by the learner
  float regret = 5;
}

// Next tag: 2
// This message stores the data generated within one trial
message OneTrialData {
  repeated DataItem data_items  = 1;
}
